src/pages/_app.tsx

```diff
@@ -20,7 +20,7 @@ const MyApp = ({ Component, pageProps }: AppProps) => {
                 content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no, viewport-fit=cover"
               />
             </Head>
-            <Layout>
+            <Layout session={pageProps.session}>
```

Comment:
I don't think this is necessarily bad but the docs seem to prefer calling useSession in the component you need access to the session. This feels like a mix of prop drilling and context.

I'd prefer calling useSession in the and displaying 'UserAvatar' if they are logged in or 'Log in' if they aren't